docs/source/user-guide/vision/VisionData.rst

```diff
@@ -0,0 +1,184 @@
+.. _vision_data_class:
+
+========================
+The Vision Data Class
+========================
+The :class:`VisionData <deepchecks.vision>` data class is the base deepchecks class for
+representing the data for a vision task. It is essentially a wrapper around a batch loader of images, labels,
+and predictions in :doc:`Deepchecks' format </user-guide/vision/supported_tasks_and_formats>`
+which is used as the input for the different :doc:`vision checks </checks_gallery/vision>`.
+In addition it stores cache regarding the data, labels and predictions it encountered during the check run.
+
+
+Class Properties
+================
+
+The common properties are:
+
+- **batch_loader** - The batch loader can be either a pytorch DataLoader, a tensorflow Dataset or any custom
+  generator. It is important to note that the data loaded by the
+  batch loader iterator must be **shuffled**.
+
+- **task_type** - The task type of the data, can be either ``classification``, ``object_detection``,
+  ``semantic_segmentation``, or ``other``. Data format validation is done upon creation of VisionData based
+  on the selected task type.
+
+- **label_map** - A dictionary mapping class ids to their names.
+
+Creating a VisionData Object
+============================
+
+From Generic Generator
+----------------------
+If you are not already using pytorch or tensorflow for the project, this is the most recommended option.
+The custom generator can be implemented in any way you like, as long as it outputs the data in the
+:doc:`required format </user-guide/vision/supported_tasks_and_formats>`
+and that it loads the data in a shuffled manner. The following is an example of a custom generator based on data that
+is loaded and stored in memory as numpy arrays.
+
+For an example of a custom generator that loads the data from disk batch by batch see the
```

Comment:
In the paragraph above you're saying "The following example is a custom generator loading from disk", and here you're saying "to see custom generator from disk go somewhere else" which confused me.